homeassistant/components/homekit_controller/config_flow.py

```diff
@@ -77,6 +82,7 @@ def __init__(self):
             key = user_input["device"]
             self.hkid = self.devices[key]["id"]
             self.model = self.devices[key]["md"]
+            await self.async_set_unique_id(normalize_hkid(self.hkid))
```

Comment:
You should set `raise_on_progress=False`, as this is the user manually configuring something.